### PR TITLE
Support passing --registry to cargo publish

### PR DESCRIPTION
**Pull request by jstoneham**
Add command-line arg and toml variable to pass a Cargo registry name (as used in the `[registries]` section of `.cargo/config`, etc.) to publish to, so it's possible to publish to alternate repositories.

Note: I actually completed an implementation of figuring out what the local index of this registry is so we could watch for an update in `wait_for_publish` - only to find that it always timed out because the local index doesn't seem to get updated.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -681,12 +681,17 @@ fn release_packages<'m>(
...
                 return Ok(103);
             }
             let timeout = std::time::Duration::from_secs(30);
-            cargo::wait_for_publish(crate_name, &base.version_string, timeout, dry_run)?;
+
+            // cargo publish doesn't update local registries that aren't crates.io
+            if pkg.config.registry().is_none() {
+                cargo::wait_for_publish(crate_name, &base.version_string, timeout, dry_run)?;
+            }
         }
```
Comment (epage) [inline src/main.rs]: Maybe make the comment a `log::debug!` message?

File: docs/reference.md [hunk 2]
```diff
@@ -48,6 +48,7 @@ Configuration is read from the following (in precedence order)
 | `exclude-paths` |               | List of path globs (`gitignore` style) for what files should be ignored when change-tracking (sub-crates are already ignored). |
 | `sign-commit`  | `--sign`        | bool   | Use GPG to sign git commits and tag generated by cargo-release. [Further information](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work) |
 | `push-remote`  | `--push-remote` | string | Default git remote to push |
+| `registry`     | `--registry`    | string | Cargo registry name to publish to (default uses Rust's default, which goes to `crates.io`) |
```
Comment (epage) [inline docs/reference.md]: Could you also update the changelog?

**Comment by epage** [comment]
Thanks! And sorry for the delay; I lost track of this in my inbox

**Comment by jstoneham** [comment]
No worries. Thanks for the merge.

Any idea when you plan to release?

**Comment by sunng87** [comment]
@jstoneham I will create a release once #204 merged.

**Comment by sunng87** [comment]
0.13.3 released with this patch

**Comment by jstoneham** [comment]
Thanks!

**Pull request merged**